src/database/core/ReadonlyRestClient.ts

```diff
@@ -98,33 +91,33 @@ export class ReadonlyRestClient {
   }
 
   /** @inheritDoc */
-  unlisten(query, tag) {
-    var listenId = ReadonlyRestClient.getListenId_(query, tag);
+  unlisten(query: Query, tag: number | null) {
+    const listenId = ReadonlyRestClient.getListenId_(query, tag);
     delete this.listens_[listenId];
   }
 
   /** @inheritDoc */
-  refreshAuthToken(token) {
+  refreshAuthToken(token: string) {
     // no-op since we just always call getToken.
   }
 
   /** @inheritDoc */
-  onDisconnectPut(pathString, data, opt_onComplete) { }
+  onDisconnectPut(pathString: string, data: any, onComplete?: (a: string, b: string) => any) { }
```

Comment:
It seems like converting ServerActions to an abstract base class with default implementations for these methods would probably remove the need for the empty methods. I don't think it should block this PR but something to do in the future.